### PR TITLE
✨ jboss/wildfly & EAP roles

### PR DESCRIPTION
**Pull request by savitharaghunathan**

### CODE REVIEW

File: ec2/roles/jboss-widlfly/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+    - hosts: servers
+      roles:
+         - { role: username.rolename, x: 42 }
+
+License
+-------
+
+BSD
```
Comment (fabianvf) [inline ec2/roles/jboss-widlfly/README.md:33]: Is this intentional?

File: ec2/roles/jboss-widlfly/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+language: python
+python: "2.7"
```
Comment (fabianvf) [inline ec2/roles/jboss-widlfly/.travis.yml:3]: do we want 2.7 for some reason? We really should be only using 3+ by this point

File: ec2/roles/jboss-eap/.travis.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+language: python
+python: "2.7"
```
Comment (fabianvf) [inline ec2/roles/jboss-eap/.travis.yml:3]: this one too

File: ec2/roles/jboss-widlfly/.travis.yml (hunk 2 above)
Comment (savitharaghunathan) [reply ec2/roles/jboss-widlfly/.travis.yml:3]: No. Its from the boilerplate code. Removing it right away. Good catch.

**Comment by fabianvf** [comment]
couple questions but the roles themselves look great

File: ec2/roles/jboss-widlfly/.travis.yml (hunk 2 above)
Comment (savitharaghunathan) [reply ec2/roles/jboss-widlfly/.travis.yml:3]: ^ meant for the bsd question. 
No. All these are boilerplate code. I will update them

File: ec2/roles/jboss-widlfly/README.md (hunk 1 above)
Comment (savitharaghunathan) [reply ec2/roles/jboss-widlfly/README.md]: Remove BSD licensing and added a line about the role

File: ec2/roles/jboss-widlfly/.travis.yml (hunk 2 above)
Comment (savitharaghunathan) [reply ec2/roles/jboss-widlfly/.travis.yml]: Updated the version to 3.11. That is what I am using but open to suggestion

File: ec2/roles/jboss-eap/.travis.yml (hunk 3 above)
Comment (savitharaghunathan) [reply ec2/roles/jboss-eap/.travis.yml]: updated to 3.11

File: ec2/README.md [hunk 4]
```diff
@@ -49,7 +49,21 @@ This directory contains ansible automation to create a single ec2 host and insta
...
+
+* Make sure to uncomment the role `jboss-wildfly` and comment `jboss-eap` in `setup_node.yml`
+
+Run `./setup_node.sh`
+
+### To create jboss EAP instance
+
+* We will be using offline installation for this role.Please [download](https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=appplatform&version=7.4) `jboss-eap-7.4.zip` from the RH Customer Portal and place it in `../ec2/jboss-eap-7.4.zip`. Make sure the file type is `zip`.
+
+* Make sure to uncomment the role `jboss-eap` and comment `jboss-wildfly` in `setup_node.yml`
```
Comment (jwmatthews) [inline ec2/README.md:63]: Small typo: s/setup_node.yml/setup_nodes.yml

File: ec2/README.md [hunk 5]
```diff
@@ -49,7 +49,21 @@ This directory contains ansible automation to create a single ec2 host and insta
...
+
+Run `./setup_node.sh`
```
Comment (jwmatthews) [inline ec2/README.md:65]: Small typo: s/setup_node.yml/setup_nodes.yml

**Comment by jwmatthews** [comment]
I'm seeing below when I try to run:

```
./setup_nodes.sh 
ERROR! the role 'jboss-wildfly' was not found in /Users/jmatthews/git/jwmatthews/playpen/ec2/roles:/Users/jmatthews/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/Users/jmatthews/git/jwmatthews/playpen/ec2

The error appears to be in '/Users/jmatthews/git/jwmatthews/playpen/ec2/setup_nodes.yml': line 14, column 7, but may
be elsewhere in the file depending on the exact syntax problem.


The offending line appears to be:


    - role: jboss-wildfly
      ^ here
```



Small typo in directory name of the role:

roles/jboss-widlfly

File: ec2/README.md [hunk 6]
```diff
@@ -49,7 +49,21 @@ This directory contains ansible automation to create a single ec2 host and insta
 ## To create the environments
 1. `cp sample.my_vars.yml my_vars.yml`
 2. `./create_infra.sh`
-3. `./setup_node.sh`
+3. 
+### To create jboss wildfly instance
+
+* Make sure to uncomment the role `jboss-wildfly` and comment `jboss-eap` in `setup_node.yml`
```
[the remark this answers is not in the document]
Comment (savitharaghunathan) [reply ec2/README.md]: updated. Thank you.

File: ec2/README.md (hunk 5 above)
Comment (savitharaghunathan) [reply ec2/README.md]: updated. Thank you.

**Comment by jwmatthews** [comment]
Small nit to consider or dismiss, just sharing what I was thinking as I played through to deploy EAP.

I hit the below, didn't fully read the docs exactly, as I didn't rename the file I downloaded to `../ec2/jboss-eap-7.4.zip`

```
TASK [middleware_automation.wildfly.wildfly_install : Validate existing zipfiles jboss-eap-7.4.zip for offline installs] *****
fatal: [ec2-34-231-110-100.compute-1.amazonaws.com]: FAILED! => {"assertion": "wildfly_version is defined and wildfly_archive_filename is exists", "changed": false, "evaluated_to": false, "msg": "An offline install was requested, but files are not present on the controller: jboss-eap-7.4.zip"}
```

I visited:
https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=appplatform&version=7.4

and Downloaded: "[Red Hat JBoss Enterprise Application Platform 7.4](https://access.redhat.com/jbossnetwork/restricted/softwareDetail.html?softwareId=99481&product=appplatform&version=7.4&downloadType=distributions)"

That downloaded a file:  "jboss-eap-7.4.0.zip"

If I followed the doc I would have renamed this to "jboss-eap-7.4.zip", then assume be OK.

Wondering what we want to do.  
 - Do we want to have a user rename the file after downloaded?
 - Do we want to assume this is just working with "7.4.0" so we update, "[wildfly_archive_filename](https://github.com/migtools/playpen/pull/2/files#diff-fa5e0820a72a789bff97df1e18090b4bfc5cf489d1bdd46b61578700411827bdR8)"
 - Do we want to set the file name default in my_vars.yml and recommend user updates that if the version changes, also add in there the other eap version vars?

**Comment by savitharaghunathan** [comment]
> Small nit to consider or dismiss, just sharing what I was thinking as I played through to deploy EAP.
> 
> I hit the below, didn't fully read the docs exactly, as I didn't rename the file I downloaded to `../ec2/jboss-eap-7.4.zip`
> 
> ```
> TASK [middleware_automation.wildfly.wildfly_install : Validate existing zipfiles jboss-eap-7.4.zip for offline installs] *****
> fatal: [ec2-34-231-110-100.compute-1.amazonaws.com]: FAILED! => {"assertion": "wildfly_version is defined and wildfly_archive_filename is exists", "changed": false, "evaluated_to": false, "msg": "An offline install was requested, but files are not present on the controller: jboss-eap-7.4.zip"}
> ```
> 
> I visited: https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=appplatform&version=7.4
> 
> and Downloaded: "[Red Hat JBoss Enterprise Application Platform 7.4](https://access.redhat.com/jbossnetwork/restricted/softwareDetail.html?softwareId=99481&product=appplatform&version=7.4&downloadType=distributions)"
> 
> That downloaded a file: "jboss-eap-7.4.0.zip"
> 
> If I followed the doc I would have renamed this to "jboss-eap-7.4.zip", then assume be OK.
> 
> Wondering what we want to do.
> 
> * Do we want to have a user rename the file after downloaded?
> * Do we want to assume this is just working with "7.4.0" so we update, "[wildfly_archive_filename](https://github.com/migtools/playpen/pull/2/files#diff-fa5e0820a72a789bff97df1e18090b4bfc5cf489d1bdd46b61578700411827bdR8)"
> * Do we want to set the file name default in my_vars.yml and recommend user updates that if the version changes, also add in there the other eap version vars?

I can definitely update that to reflect the filename to jboss-eap-7.4.0.zip. I do want to reorganize the roles and use vars, may be combine them into one if possible. I am planning to do that in the next pass. Another issue I ran into was the downloaded zip file from the RH customer portal did not extract properly in the remote server.

**Comment by jwmatthews** [comment]
I hit below error but unsure right now why, this was from a re-run after I ran EAP first, so maybe that is related.

```

TASK [middleware_automation.wildfly.wildfly_install : Extract files from /opt/wildfly//wildfly-22.0.1.Final.zip into /opt/wildfly/.] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleError: Failed to get information on remote file (/opt/wildfly/): Permission denied
fatal: [ec2-34-231-110-100.compute-1.amazonaws.com]: FAILED! => {"changed": false, "msg": "Failed to get information on remote file (/opt/wildfly/): Permission denied"}
```

**Comment by savitharaghunathan** [comment]
@jwmatthews I made the changes we talked about. PTAL :) I am also displaying some output that basically gets localhost:8080 after the service has been started successfully.
 Also, the permission denied error is weird. if you are using the same server, the only issue it would have is port 8080 being in use. I will see if I can reproduce.

**Review by jwmatthews — APPROVED** [review]
LGTM, thank you @savitharaghunathan 

I ran through the latest changes and had success and also saw the test output at end to confirm service is running.

**Comment by jwmatthews** [comment]
> @jwmatthews I made the changes we talked about. PTAL :) I am also displaying some output that basically gets localhost:8080 after the service has been started successfully. Also, the permission denied error is weird. if you are using the same server, the only issue it would have is port 8080 being in use. I will see if I can reproduce.

Yes, makes sense that as I re-ran after a prior install on same VM there was a port conflict.

**Pull request merged**